src/condor_scripts/condor_upgrade_check

```diff
@@ -550,6 +550,179 @@ def warn_user_records():
+        format_print(
+            "Are deprecated and will be removed for V24 of HTCondor.", offset=4
+        )
+        format_print(
+            "The new configuration syntax for the job router is defined", offset=4
+        )
+        format_print(
+            "using JOB_ROUTER_ROUTE_NAMES and JOB_ROUTER_ROUTE_<name>.", offset=4
+        )
+        format_print(
+            """For more information visit:
+  https://htcondor.readthedocs.io/en/latest/grid-computing/job-router.html?highlight=job%20router#the-htcondor-job-router""",
+            offset=4,
+        )
+        format_print(
+            "Note: The removal will occur during the lifetime of the",
+            offset=4,
+            newline=True,
+        )
+        format_print("      HTCondor V23 feature series.", offset=4)
+        return 1
+    return 0
+
+
+def check_desktop_policy():
+    """Check for use of desktop policy configuration knobs that became hidden in V10.4.0"""
+    htc_version = htcondor.param["CONDOR_VERSION"].split(".")
+    if int(htc_version[0]) < 10 or (
+        int(htc_version[0]) == 10 and int(htc_version[1]) < 4
+    ):
+        # Config items formally avaible 100% of the time
+        attrs = [
+            "ActivationTimer",
+            "ConsoleBusy",
+            "CpuBusy",
+            "CpuIdle",
+            "JustCPU",
+            "KeyboardBusy",
+            "KeyboardNotBusy",
+            "LastCkpt",
+            "MachineBusy",
+            "NonCondorLoadAvg",
+        ]
+        default_suspend_line = "SUSPEND = ($(KeyboardBusy) || ( (CpuBusyTime > 120) && $(ActivationTimer) > 90))"
+        found_attrs = []
+        has_cpu_busy_time = []
+
+        p = subprocess.run(["condor_config_val", "-dump"], stdout=subprocess.PIPE)
+        cmd_output = p.stdout.rstrip().decode()
+        has_problem = False
+        for line in cmd_output.split("\n"):
+            line = line.strip()
+            check = line.lower()
+            parts = check.split("=", 1)
+            if len(parts) < 2:
+                continue
+            for attr in attrs:
+                if attr.lower() in parts[1]:
+                    if attr == "CpuBusy":
+                        pos = check.find("cpubusy")
+                        false_positive = True
+                        while pos != -1:
+                            if check.find("cpubusytime", pos) != pos:
+                                false_positive = False
+                                break
+                            pos = check.find("cpubusy", pos + 1)
+                        if false_positive:
+                            continue
+                    not_in_other_def = True
+                    for other in attrs:
+                        if other.lower() == parts[0].strip():
+                            not_in_other_def = False
+                            break
+                    if not_in_other_def:
+                        found_attrs.append(attr)
```

Comment:
Using a set for `found_attrs` seems more appropriate as we only care to know if they're using a forbidden `attr` at least once and don't want to print out the same `attr` mutliple times later.
```suggestion
        found_attrs = set()
        has_cpu_busy_time = []

        p = subprocess.run(["condor_config_val", "-dump"], stdout=subprocess.PIPE)
        cmd_output = p.stdout.rstrip().decode()
        has_problem = False
        for line in cmd_output.split("\n"):
            line = line.strip()
            check = line.lower()
            parts = check.split("=", 1)
            if len(parts) < 2:
                continue
            for attr in attrs:
                if attr.lower() in parts[1]:
                    if attr == "CpuBusy":
                        pos = check.find("cpubusy")
                        false_positive = True
                        while pos != -1:
                            if check.find("cpubusytime", pos) != pos:
                                false_positive = False
                                break
                            pos = check.find("cpubusy", pos + 1)
                        if false_positive:
                            continue
                    not_in_other_def = True
                    for other in attrs:
                        if other.lower() == parts[0].strip():
                            not_in_other_def = False
                            break
                    if not_in_other_def:
                        found_attrs.add(attr)
```

Most other methods used later (`found_attrs.clear()`, `found_attrs.remove()`) work the same for sets, except for `found_attrs.sort()`.